### PR TITLE
Implemented Player#getEffectivePermissions on Bukkit and Nukkit

### PR DESCRIPTION
**Pull request by McRuben**

### CODE REVIEW

**Comment by manuelkollus** [comment]
I would add null checks in that case, what do you want to do if the IPermissble class is null? 

Also, you are not using the latest code style currently used on CloudNet. In retrospect it doesn't look the same anymore.

Yours sincerely,
Manuel Kollus

File: cloudnet-driver/src/main/java/de/dytanic/cloudnet/driver/permission/IPermissionManagement.java [hunk 1]
```diff
@@ -271,4 +271,29 @@ default boolean tryExtendedGroups(IPermissionGroup permissionGroup, String group
 
         return false;
     }
+
+    default Collection<Permission> getAllPermissions(IPermissible permissible) {
```
Comment (manuelkollus) [inline cloudnet-driver/src/main/java/de/dytanic/cloudnet/driver/permission/IPermissionManagement.java]: Why no null checks

File: cloudnet-driver/src/main/java/de/dytanic/cloudnet/driver/permission/IPermissionManagement.java [hunk 2]
```diff
@@ -271,4 +271,29 @@ default boolean tryExtendedGroups(IPermissionGroup permissionGroup, String group
...
+        return this.getAllPermissions(permissible, null);
+    }
+
+    default Collection<Permission> getAllPermissions(IPermissible permissible, String group) {
+        if (permissible == null)
+            return null;
```
Comment (manuelkollus) [inline cloudnet-driver/src/main/java/de/dytanic/cloudnet/driver/permission/IPermissionManagement.java]: You can get an empty collection back.

File: cloudnet-driver/src/main/java/de/dytanic/cloudnet/driver/permission/IPermissionManagement.java [hunk 3]
```diff
@@ -271,4 +271,29 @@ default boolean tryExtendedGroups(IPermissionGroup permissionGroup, String group
...
+
+        Collection<Permission> permissions = new ArrayList<>(permissible.getPermissions());
```
Comment (manuelkollus) [inline cloudnet-driver/src/main/java/de/dytanic/cloudnet/driver/permission/IPermissionManagement.java]: You can underline an line break

**Comment by realPanamo** [comment]
About the code style not matching: Brackets at the next line is what neither the maintainers nor the most of the participants use or want. So it's fine in my opinion.

**Comment by realPanamo** [comment]
I agree with the null checks

File: cloudnet-driver/src/main/java/de/dytanic/cloudnet/driver/permission/IPermissionManagement.java (hunk 2 above)
Comment (GiantTreeLP) [reply cloudnet-driver/src/main/java/de/dytanic/cloudnet/driver/permission/IPermissionManagement.java]: Yeah, returning an empty collection is the only correct answer here.

Comment (McRuben) [reply cloudnet-driver/src/main/java/de/dytanic/cloudnet/driver/permission/IPermissionManagement.java]: fixed

**Comment by realPanamo** [comment]
Please use the Google code style brought in by #3

**Pull request merged**